### DISCUSSION TITLE
How does Typetta achieve typing for it's security middleware?

### DISCUSSION THREAD

**Discussion by Atlinx** [twinlogix/typetta#231] [Q&A] [answered]
Is typing done through generics, code generation, or a mix of both?

**Comment by edobrb** [comment, marked as answer]
It's a mix of both. Some types are generated in order to complete the various [DAOGenerics](https://github.com/twinlogix/typetta/blob/53a22eb5d49184eb91d8a27f007df7c364a54e46/src/dal/dao/dao.types.ts#L128-L188), then these are used by Typetta.

This is inside the generated code:
https://github.com/twinlogix/typetta/blob/53a22eb5d49184eb91d8a27f007df7c364a54e46/tests/security/dao.mock.ts#L823-L843
https://github.com/twinlogix/typetta/blob/53a22eb5d49184eb91d8a27f007df7c364a54e46/tests/security/dao.mock.ts#L899

And this is inside the internals:
https://github.com/twinlogix/typetta/blob/53a22eb5d49184eb91d8a27f007df7c364a54e46/src/dal/dao/middlewares/securityPolicy/security.types.ts#L27-L36
